homeassistant/components/shelly/binary_sensor.py

```diff
@@ -0,0 +1,69 @@
+"""Binary sensor for Shelly."""
+import aioshelly
+
+from homeassistant.components import binary_sensor
+from homeassistant.const import STATE_OFF, STATE_ON
+from homeassistant.helpers.entity import Entity
+
+from . import ShellyBlockEntity, ShellyDeviceWrapper
+from .const import DOMAIN
+
+SENSORS = {
+    "dwIsOpened": binary_sensor.DEVICE_CLASS_OPENING,
+    "flood": binary_sensor.DEVICE_CLASS_MOISTURE,
+    "overpower": None,
+    "smoke": binary_sensor.DEVICE_CLASS_SMOKE,
+}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up sensors for device."""
+    wrapper = hass.data[DOMAIN][config_entry.entry_id]
+    sensors = []
+
+    for block in wrapper.device.blocks:
+        for attr in SENSORS:
+            if not hasattr(block, attr):
+                continue
+
+            sensors.append(ShellySensor(wrapper, block, attr))
+
+    if sensors:
+        async_add_entities(sensors)
+
+
+class ShellySensor(ShellyBlockEntity, Entity):
```

Comment:
This should inherit `BinarySensorEntity`.

https://github.com/home-assistant/core/blob/22a123fd4b9d8dc41fe2de46bea3a02154ed9c58/homeassistant/components/binary_sensor/__init__.py#L147

Reply:
What a stupid mistake 🤦 Fixed.